Automattic/jetpack
# Deprecation notices: Allow functionality to be used, and update styles

Fixes https://github.com/Automattic/vulcan/issues/511


## Proposed changes:

* This PR has two purposes: ensure deprecation notices are easily extendable if there are 1 or more notices to add, and 
modify notice styling depending on design conversation outcomes. 
* In terms of styling, the My Jetpack and WP Admin notices will be consistent with style changes, and the Jetpack Settings / Dashboard notices will stay the same until a more complete overhaul of notices is done. Text changes should apply though, which is generally more relevant to the examples given.
* Additional changes include also updating the cookie settings to ensure they work in more environments (removing `cookie.serialize`), and ensures the WP admin notice only appears on the main WP admin dashboard page.

### Other information:


## Jetpack product discussion

* Design conversation: p1HpG7-uu9-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

To test this out, you will need to add in a new notice. This will all be documented in a post once this PR is completed and merged.

* To add the code needed for testing, set the `show` property to `true` in `$this->notices` within the `__construct` function in `class-deprecate.php`.
* Make sure to build both Jetpack and My Jetpack: `jetpack build plugins/jetpack`, `jetpack build packages/my-jetpack`

Once the code is added, apply this PR locally, and make sure to build Jetpack and the My Jetpack package. Then, test the following areas:
* Jetpack Settings page and Dashboard (only those two pages, not other pages such as`/wp-admin/admin.php?page=jetpack_modules`).
* WP Admin (only on `wp-admin/index.php`)
* My Jetpack

**Jetpack Settings / Dashboard:**

In this case we're not testing design changes, but rather checking functionality / code changes and the suggested default text.

<img width="1051" alt="Screenshot 2024-10-16 at 09 55 33" src="https://github.com/user-attachments/assets/0b43d548-0c93-4045-bf42-b9a6e9e9b83c">


**My Jetpack:**

<img width="1101" alt="Screenshot 2024-10-09 at 16 35 33" src="https://github.com/user-attachments/assets/f6051b36-d7b3-4f86-8c91-894e721909b4">

**WP Admin Dashboard:**

<img width="1839" alt="Screenshot 2024-10-09 at 16 35 24" src="https://github.com/user-attachments/assets/f54639ff-1153-46a6-a15a-1ea36c6b1016">
